doofinder/includes/api/class-store-api.php

```diff
@@ -177,10 +249,138 @@ public function get_product_datatype()
                     ]
                 ]
             ]
         ];
     }
+
+    private function get_callback_urls($api_keys, $primary_language)
+    {
+        $callback_urls = [];
+        $currency = 'EUR';
+        foreach ($api_keys as $item) {
+            $code = $item['lang']['locale'] ?? $item['lang']['code'] ?? $primary_language;
+            $lang = Helpers::get_language_from_locale($code);
+            $code = Helpers::format_locale_to_hyphen($code);
+            $callback_urls[$code][$currency] = self::build_callback_url(
+                $this->language->get_home_url($lang),
+                '/wp-json/doofinder/v1/index-status/?token=' . $this->api_key
+            );
+        }
+        return $callback_urls;
+    }
+
+    /**
+     * This method takes the base url and adds
+     *
+     * @param [type] $base_url
+     * @param [type] $endpoint_path
+     * @return void
+     */
+    private static function build_callback_url($base_url, $endpoint_path)
+    {
+        $parsed_url = parse_url($base_url);
+        parse_str($parsed_url['query'], $parameters);
+        $callback_url = $parsed_url['scheme'] . '://' . $parsed_url['host'] . rtrim($parsed_url['path'], '/') . '/' . ltrim($endpoint_path, '/');
+
+        // Combine any existing parameters with any possible endopoint path parameters
+        if (!empty($parameters)) {
+            parse_str(parse_url($callback_url, PHP_URL_QUERY), $endpoint_parameters);
+            $combined_parameters = array_merge($parameters, $endpoint_parameters);
+            $callback_url = strtok($callback_url, '?');
+            $callback_url .= '?' . http_build_query($combined_parameters);
+        }
+
+        return $callback_url;
+    }
+
+    /**
+     * Generates an api_password and returns the store options.
+     *
+     * @return void
+     */
+    private static function get_store_options()
+    {
+        $password_data = self::create_application_credentials();
+        if (!is_null($password_data)) {
+            return [
+                "url" => get_bloginfo('url'),
+                'api_pass' => $password_data['api_pass'],
+                'api_user' => $password_data['api_user']
+            ];
+        }
+        return NULL;
+    }
+
+    /**
+     * This method checks if there is an application password set.
+     *
+     * @return boolean
+     */
+    public static function has_application_credentials()
+    {
+        return WP_Application_Passwords::application_name_exists_for_user(get_current_user_id(), 'doofinder');
+    }
+
+    /**
+     * Creates a new application password.
+     * If a password exists, it deletes it and creates a new password.
+     *
+     * We store the user_id and the uuid in order to know which application
+     * password we must delete.
+     *
+     * @return array Array containing api_user and api_pass
+     */
+    public static function create_application_credentials()
```

Comment:
Isn't it dangerous making this function static and public?

Reply:
Yes, I've corrected it to be private